Add docs for `Building a collection` (#83)
ansible/ansible-dev-tools: docs/getting-started/building-collection.md

# Building Ansible Collection using using Ansible Development Tools (ADT)

This guide illustrates a comprehensive Ansible development workflow that showcases the integration of various tools within the Ansible ecosystem to create an Ansible collection. The tools featured in this workflow include:

- [ansible-creator](https://github.com/ansible/ansible-creator)

> - [ansible-development-environment (pip4a)](https://github.com/ansible/ansible-development-environment)

Reviewer: ```suggestion
- [ansible-dev-environment](https://github.com/ansible/ansible-dev-environment)
```

- [VS Code Ansible extension](https://github.com/ansible/vscode-ansible)

## Step 1: Scaffolding collection using ansible-creator

- Open VS Code and click the Ansible icon in the activity bar to access Ansible Creator section. Click "Get Started" under that section to open the menu page of Ansible Creator in VS Code.

- Check system requirements and install ansible-creator if needed. Ensure all requirements in the `system requirements` box have green ticks.

![ansible-creator-menu](../media/ansible-creator-menu-installed.png)

- Click **Initialize a collection** to open the **Init** interface. Fill in the form with the collection name, initialization path, verbosity, and logging options.

- Click **Create** to scaffold the collection in the desired location. You have the option to review the logs or open the log file in VS Code editor for details.

- Click the **Open collection** button to add the collection folder to the workspace.

![scaffold-collection](../media/create-collection.gif)

NOTE: For a more detailed explanation about using Ansible Creator in the VS Code Ansible Extension, refer to [doc: ansible-creator].

> ## Step 2: Installing your collection using ansible-development-environment (ade, pip4a)

Reviewer: ```suggestion
## Step 2: Installing your collection using ansible-development-environment (ade)
```

> - With the initial collection structure in place, use 'ansible-development-environment' (pip4a) to install the newly created collection in editable mode, similar to Python modules.

Reviewer: ```suggestion
- With the initial collection structure in place, use 'ansible-dev-environment' (ade) to install the newly created collection in editable mode, similar to Python modules.
```

- Navigate to the collection directory and run the following command:

```console

> $ pip4a install -e .

Reviewer: ```suggestion
$ ade install -e .
```

```

- This installation method adds the collection in the system paths, enabling Ansible to recognize it. Additionally, it improves the development process by allowing on-the-go changes to the module code.

- You can check if the collection is installed or not by using `ansible-galaxy` command. Run the following command in a terminal to list installed collections. Your newly created collection will appear in the list.

```console
$ ansible-galaxy collection list
```

> ![pip4-a](../media/pip4a-installation.gif)

Reviewer: ```suggestion
![ade](../media/ade-installation.gif)
```

## Step 3: Add python code to bring the collection to life

- Navigate to the collection directory and navigate to `plugins/modules/`. Create a [module-name].py file and add documentation, examples and logic to the module.

> - Because you installed using pip4a, you can change module code dynamically and observe the effects during playbook execution.

Reviewer: ```suggestion
- Because you installed using ADE, you can change module code dynamically and observe the effects during playbook execution.
```
